pkgs/misc/bergamot-models/default.nix

```diff
@@ -0,0 +1,32 @@
+{ lib, stdenv, fetchurl }:
+
+let
+  modelSpecs = (builtins.fromJSON (builtins.readFile ./models.json)).models;
+  withCodeAsKey = f: { code, ... }@attrs: lib.nameValuePair code (f attrs);
+  mkModelPackage = { name, code, version, url, checksum, ... }:
+    stdenv.mkDerivation {
+      pname = "bergamot-model-${code}";
+      version = toString version;
+
+      src = fetchurl {
+        inherit url;
+        sha256 = checksum;
+      };
+
+      installPhase = ''
+        TARGET="$out/share/bergamot/models"
+        mkdir -p "$TARGET"
+        tar -xzf "$src" -C "$TARGET"
+      '';
+
+      meta = {
+        description = "Bergamot machine translation model - ${name}";
+        homepage = "https://browser.mt/";
+        # https://github.com/browsermt/students/blob/master/LICENSE.md
+        license = lib.licenses.cc-by-sa-40;
+      };
+    };
+
+in lib.listToAttrs (map (withCodeAsKey mkModelPackage) modelSpecs) // {
+  passthru.updateScript = ./update.sh;
```

Comment:
(I'm not sure whether this is the right way to set an update script for a whole
package set, or whether this is possible at all).